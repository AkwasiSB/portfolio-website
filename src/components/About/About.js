import React from 'react';
import { IoLogoFacebook, IoLogoTwitter, IoLogoLinkedin } from 'react-icons/io';
import { FaAngellist, FaGithub, FaEnvelope } from 'react-icons/fa';
import profileImage from '../../images/profile-image.png';
import styles from './About.scss';

export default class About extends React.Component {
	constructor(props) {
		super(props);

		this.mySkills = ['C/GTK+', 'HTML & CSS', 'Javascript', 'HTML5 & CSS3', 'Git', 'Bootstrap', 'React.js', 'React-Native', 'Express.js', 'ES6', 'GraphQL/Apollo', 'Redux', 'Webpack', 'Babel', 'Yarn', 'npm', 'SCSS/SASS', 'ESLint', 'Jest', 'TypeScript', 'Enzyme', 'TDD', 'BEM', ' Node.js', 'WebRTC', 'SQL/MySQL', 'SQLite', 'JSON/XML', 'jQuery', 'Cordova', 'SinonJS', 'AJAX'];
	}

	renderSkills = () => (
		this.mySkills.map((skill, index) => (
			<div key={index} className={styles.skill}>{skill}</div>
		))
	)

	render() {
		return (
			<section id='about' className={styles.about}>
				<h2>ABOUT ME</h2>

				<div className={styles.aboutContainer}>
					<div className={styles.aboutDetails}>
						<div className={styles.aboutProfile}>
							<img className={styles.aboutPhoto} src={profileImage} alt='' />

							<h3>Akwasi Sefa-Boakye</h3>

							<p><strong>Frontend Developer</strong></p>

							<div className={styles.socialContainer}>
								<a
									href='mailto:akwasisefaboakye@yahoo.com'
									target='_blank'
									rel='noopener noreferrer'
									className={`${styles.socialLink} -yahoo`}>
									<FaEnvelope size={20} />
								</a>

								<a
									href='https://www.facebook.com/akwasisefaboakye'
									target='_blank'
									rel='noopener noreferrer'
									className={`${styles.socialLink} -facebook`}>
									<IoLogoFacebook size={20} />
								</a>

								<a
									href='https://twitter.com/kwasisefaboakye'
									target='_blank'
									rel='noopener noreferrer'
									className={`${styles.socialLink} -twitter`}>
									<IoLogoTwitter size={20} />
								</a>

								<a
									href='https://linkedin.com/in/akwasisefaboakye'
									target='_blank'
									rel='noopener noreferrer'
									className={`${styles.socialLink} -linkedin`}>
									<IoLogoLinkedin size={20} />
								</a>

								<a
									href='https://angel.co/akwasi-sefa-boakye'
									target='_blank'
									rel='noopener noreferrer'
									className={`${styles.socialLink} -angellist`}>
									<FaAngellist size={20} />
								</a>

								<a
									href='https://github.com/AkwasiSB'
									target='_blank'
									rel='noopener noreferrer'
									className={`${styles.socialLink} -github`}>
									<FaGithub size={20} />
								</a>
							</div>
						</div>

						<div className={styles.aboutInfo}>
							<p>Hi, I’m Akwasi Sefa-Boakye. I’m passionate about what I do, and I love to help people. Nothing is more fulfilling than being part of a team with similar interests, and an organization that values its employees.</p>

							<p>Frontend Developer and Founder with the passion to drive projects to success</p>
						</div>
					</div>

					<div className={styles.aboutSkills}>
						<h3>Skills</h3>

						<div className={styles.skillsContainer}>{this.renderSkills()}</div>
					</div>
				</div>
			</section>
		);
	}
}
