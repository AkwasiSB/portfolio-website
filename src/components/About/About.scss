$screen-sm: 768px;
$screen-md: 1200px;

.about {
	display: flex;
	flex-direction: column;
	padding: 6rem 2rem 0;

	@media (min-width: $screen-sm) {
		h2 {
			padding-left: 13%;
		}
	}
}

.about-container {
	padding: 2rem 1rem;
}

.about-details {
	position: relative;
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;

	@media (min-width: $screen-sm) {
		flex-direction: row;
		align-items: flex-start;
	}
}

.about-profile {
	position: relative;
	display: flex;
	flex-direction: column-reverse;
	align-items: center;
	border-radius: 2rem;

	@media (min-width: $screen-sm) {
		flex-direction: column;
		padding: 0.4rem 5rem;
		margin-left: 9rem;
	}

	h3 {
		font-size: 1.5rem;
		color: #555;
		font-weight: 400;
		margin-top: 0;
		margin-bottom: 1.5rem;
		
		@media (min-width: $screen-sm) {
			margin-top: 1.5rem;
			margin-bottom: 0;
		}
	}

	.about-photo {
		display: block;
		width: 15rem;
		height: 15rem;
		border-radius: 50%;
		border: 3px dashed rgb(128, 0, 0);
		margin-top: 1.5rem;
		margin-bottom: 6rem;
		background-color: #fff;

		@media (min-width: $screen-sm) {
			width: 20rem;
			height: 20rem;
			margin-top: 0;
			margin-bottom: 0;
		}
	}
}

.social-container {
	position: absolute;
	bottom: 0;
	left: 50%;
	transform: translateX(-50%);
	display: flex;
	
	@media (min-width: $screen-sm) {
		top: -5%;
		left: -6rem;
		flex-direction: column;
	}
}

:local(.social-link) {
	display: inline-block;
	width: 4rem;
	height: 4rem;
	border-radius: 2rem;
	text-align: center;
	line-height: 4.8rem;
	margin-right: 1rem;
	margin-top: 0;
	color: #fff;
	box-shadow: 3px 3px 13px 5px rgba(0, 0, 0, 0.45);

	&:last-child {
		margin-right: 0;
	}

	:global &.-yahoo {
		background-color: #5657bb;
	}

	:global &.-facebook {
		background-color: #3b58a4;
	}

	:global &.-twitter {
		background-color: #1da1f2;
	}

	:global &.-linkedin {
		background-color: #0077b5;
	}

	:global &.-angellist {
		color: #333;
		background-color: #fff;
		border: 0.05rem solid #888;
	}

	:global &.-github {
		background-color: #222;
	}

	@media (min-width: $screen-sm) {
		margin-right: 0;
		margin-top: 1rem;
		background-color: #888 !important;
		transition-property: background-color 250ms ease-in;
		box-shadow: none;

		:global &.-angellist,
		:global &.-github {
			color: #fff;
		}

		:global &.-yahoo:hover {
			background-color: #5657bb !important;
		}

		:global &.-facebook:hover {
			background-color: #3b58a4 !important;
		}

		:global &.-twitter:hover {
			background-color: #1da1f2 !important;
		}

		:global &.-linkedin:hover {
			background-color: #0077b5 !important;
		}

		:global &.-angellist:hover {
			color: #333 !important;
			background-color: #fff !important;
			border: 0.05rem solid #888;
		}

		:global &.-github:hover {
			background-color: #222 !important;
		}
	}
}

.about-info {
	position: relative;
	font-size: 2rem;
	font-style: oblique;
	color: #555;
	line-height: 2.1rem;
	padding: 4rem 3rem;
	border-radius: 2rem;
	margin-top: 3rem;
	background-color: #fff;
	box-shadow: -3px 5px 10px 0 rgba(0, 0, 0, 0.2);

	@media (min-width: $screen-sm) {
		width: 40%;
		margin-top: 0;
		box-shadow: 3px -5px 12px 0 rgba(0, 0, 0, 0.2);
	}

	&:after {
		content: "";
		width: 0;
		position: absolute;
		top: -1.5rem;
		left: 50%;
		transform: translateX(-50%);
		border-left: 1.5rem solid transparent;
		border-right: 1.5rem solid transparent;
		border-bottom: 1.5rem solid  #fff;
		border-top: none;

		@media (min-width: $screen-sm) {
			top: 25%;
			left: -1.5rem;
			transform: translateX(0);
			border-top: 1.5rem solid transparent;
			border-right: 1.5rem solid #fff;
			border-bottom: 1.5rem solid transparent;
			border-left: none;
		}
	}
}

.about-skills {
	display: flex;
	flex-direction: column;
	align-items: center;
	margin-top: 2rem;
}

.skills-container {
	display: flex;
	justify-content: center;
	flex-wrap: wrap;

	@media (min-width: $screen-sm) {
		width: 700px;
	}

	.skill {
		display: flex;
		justify-content: center;
		align-items: center;
		font-size: 1.4rem;
		width: auto;
		padding-left: 1rem;
		padding-right: 1rem;
		height: 4rem;
		border-radius: 15%;
		margin: 1rem 1.5rem;
		box-shadow: 0 0 8px 3px rgba(0, 0, 0, 0.2);

		&:nth-child(6n + 1) {
			color: #fff;
			background-color: rgba(228, 101, 114, 0.7);
		}

		&:nth-child(6n + 2) {
			color: #fff;
			background-color: rgba(0, 0, 0, 0.7);
		}

		&:nth-child(6n + 3) {
			color: #fff;
			background-color: rgba(128, 0, 128, 0.8);
		}

		&:nth-child(6n + 4) {
			color: #333;
			background-color: #fff;
		}

		&:nth-child(6n + 5) {
			color: #000;
			background-color: #c0a283;
		}
	}
}
