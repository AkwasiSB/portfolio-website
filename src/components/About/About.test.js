import About from './About';

it('should render <About /> correctly', () => {
    const aboutWrapper = shallow(<About />);

    expect(aboutWrapper).toMatchSnapshot();
});
