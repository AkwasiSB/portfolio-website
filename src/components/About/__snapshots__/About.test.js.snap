// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <About /> correctly 1`] = `
<section
  className="about"
  id="about"
>
  <h2>
    ABOUT ME
  </h2>
  <div
    className="aboutContainer"
  >
    <div
      className="aboutDetails"
    >
      <div
        className="aboutProfile"
      >
        <img
          alt=""
          className="aboutPhoto"
          src=""
        />
        <h3>
          Akwasi Sefa-Boakye
        </h3>
        <p>
          <strong>
            Frontend Developer
          </strong>
        </p>
        <div
          className="socialContainer"
        >
          <a
            className="socialLink -yahoo"
            href="mailto:akwasisefaboakye@yahoo.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <FaEnvelope
              size={20}
            />
          </a>
          <a
            className="socialLink -facebook"
            href="https://www.facebook.com/akwasisefaboakye"
            rel="noopener noreferrer"
            target="_blank"
          >
            <IoLogoFacebook
              size={20}
            />
          </a>
          <a
            className="socialLink -twitter"
            href="https://twitter.com/kwasisefaboakye"
            rel="noopener noreferrer"
            target="_blank"
          >
            <IoLogoTwitter
              size={20}
            />
          </a>
          <a
            className="socialLink -linkedin"
            href="https://linkedin.com/in/akwasisefaboakye"
            rel="noopener noreferrer"
            target="_blank"
          >
            <IoLogoLinkedin
              size={20}
            />
          </a>
          <a
            className="socialLink -angellist"
            href="https://angel.co/akwasi-sefa-boakye"
            rel="noopener noreferrer"
            target="_blank"
          >
            <FaAngellist
              size={20}
            />
          </a>
          <a
            className="socialLink -github"
            href="https://github.com/AkwasiSB"
            rel="noopener noreferrer"
            target="_blank"
          >
            <FaGithub
              size={20}
            />
          </a>
        </div>
      </div>
      <div
        className="aboutInfo"
      >
        <p>
          Hi, I’m Akwasi Sefa-Boakye. I’m passionate about what I do, and I love to help people. Nothing is more fulfilling than being part of a team with similar interests, and an organization that values its employees.
        </p>
        <p>
          Frontend Developer and Founder with the passion to drive projects to success
        </p>
      </div>
    </div>
    <div
      className="aboutSkills"
    >
      <h3>
        Skills
      </h3>
      <div
        className="skillsContainer"
      >
        <div
          className="skill"
          key="0"
        >
          C/GTK+
        </div>
        <div
          className="skill"
          key="1"
        >
          HTML & CSS
        </div>
        <div
          className="skill"
          key="2"
        >
          Javascript
        </div>
        <div
          className="skill"
          key="3"
        >
          HTML5 & CSS3
        </div>
        <div
          className="skill"
          key="4"
        >
          Git
        </div>
        <div
          className="skill"
          key="5"
        >
          Bootstrap
        </div>
        <div
          className="skill"
          key="6"
        >
          React.js
        </div>
        <div
          className="skill"
          key="7"
        >
          React-Native
        </div>
        <div
          className="skill"
          key="8"
        >
          Express.js
        </div>
        <div
          className="skill"
          key="9"
        >
          ES6
        </div>
        <div
          className="skill"
          key="10"
        >
          GraphQL/Apollo
        </div>
        <div
          className="skill"
          key="11"
        >
          Redux
        </div>
        <div
          className="skill"
          key="12"
        >
          Webpack
        </div>
        <div
          className="skill"
          key="13"
        >
          Babel
        </div>
        <div
          className="skill"
          key="14"
        >
          Yarn
        </div>
        <div
          className="skill"
          key="15"
        >
          npm
        </div>
        <div
          className="skill"
          key="16"
        >
          SCSS/SASS
        </div>
        <div
          className="skill"
          key="17"
        >
          ESLint
        </div>
        <div
          className="skill"
          key="18"
        >
          Jest
        </div>
        <div
          className="skill"
          key="19"
        >
          TypeScript
        </div>
        <div
          className="skill"
          key="20"
        >
          Enzyme
        </div>
        <div
          className="skill"
          key="21"
        >
          TDD
        </div>
        <div
          className="skill"
          key="22"
        >
          BEM
        </div>
        <div
          className="skill"
          key="23"
        >
           Node.js
        </div>
        <div
          className="skill"
          key="24"
        >
          WebRTC
        </div>
        <div
          className="skill"
          key="25"
        >
          SQL/MySQL
        </div>
        <div
          className="skill"
          key="26"
        >
          SQLite
        </div>
        <div
          className="skill"
          key="27"
        >
          JSON/XML
        </div>
        <div
          className="skill"
          key="28"
        >
          jQuery
        </div>
        <div
          className="skill"
          key="29"
        >
          Cordova
        </div>
        <div
          className="skill"
          key="30"
        >
          SinonJS
        </div>
        <div
          className="skill"
          key="31"
        >
          AJAX
        </div>
      </div>
    </div>
  </div>
</section>
`;
